homeassistant/components/foscam/camera.py

```diff
@@ -90,85 +87,90 @@
-
-        if not entity_ids:
-            return
-
-        _LOGGER.debug("Moving '%s' camera(s): %s", movement, entity_ids)
+    config_new = {
+        CONF_NAME: config[CONF_NAME],
+        CONF_HOST: config["ip"],
+        CONF_PORT: config[CONF_PORT],
+        CONF_USERNAME: config[CONF_USERNAME],
+        CONF_PASSWORD: config[CONF_PASSWORD],
+        CONF_STREAM: "Main",
+    }
 
-        all_cameras = hass.data[FOSCAM_DATA][FOSCAM_ENTITIES]
-        target_cameras = [
-            camera for camera in all_cameras if camera.entity_id in entity_ids
-        ]
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config_new
+        )
+    )
 
-        for camera in target_cameras:
-            await camera.async_perform_ptz(movement, travel_time)
 
-    hass.services.async_register(
-        FOSCAM_DOMAIN, SERVICE_PTZ, async_handle_ptz, schema=SERVICE_PTZ_SCHEMA
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add a Foscam IP camera from a config entry."""
+    platform = entity_platform.current_platform.get()
+    platform.async_register_entity_service(
+        SERVICE_PTZ, SERVICE_PTZ_SCHEMA, "async_perform_ptz"
     )
 
     camera = FoscamCamera(
-        config[CONF_IP],
-        config[CONF_PORT],
-        config[CONF_USERNAME],
-        config[CONF_PASSWORD],
+        config_entry.data[CONF_HOST],
+        config_entry.data[CONF_PORT],
+        config_entry.data[CONF_USERNAME],
+        config_entry.data[CONF_PASSWORD],
         verbose=False,
     )
 
-    rtsp_port = config.get(CONF_RTSP_PORT)
-    if not rtsp_port:
-        ret, response = await hass.async_add_executor_job(camera.get_port_info)
-
-        if ret == 0:
-            rtsp_port = response.get("rtspPort") or response.get("mediaPort")
-
-    ret, response = await hass.async_add_executor_job(camera.get_motion_detect_config)
-
-    motion_status = False
-    if ret != 0 and response == 1:
-        motion_status = True
-
-    async_add_entities(
-        [
-            HassFoscamCamera(
-                camera,
-                config[CONF_NAME],
-                config[CONF_USERNAME],
-                config[CONF_PASSWORD],
-                rtsp_port,
-                motion_status,
-            )
-        ]
-    )
+    async_add_entities([HassFoscamCamera(camera, config_entry)])
 
 
 class HassFoscamCamera(Camera):
     """An implementation of a Foscam IP camera."""
 
-    def __init__(self, camera, name, username, password, rtsp_port, motion_status):
+    def __init__(self, camera, config_entry):
         """Initialize a Foscam camera."""
         super().__init__()
 
         self._foscam_session = camera
-        self._name = name
-        self._username = username
-        self._password = password
-        self._rtsp_port = rtsp_port
-        self._motion_status = motion_status
+        self._name = config_entry.title
```

Comment:
Just a reflection. If you would store the config_entry in the object you could potentially allow the entity to change any config variable in runtime.

Reply:
Yeah, no need to change anything now. If you identify this could happen it would be a perfect new PR :)